MISC: Several typo fix in corporation modals

cigarmemr opened the pull request:
Missing space between sentences in PurchaseMaterialModal when Smart Supply is enabled:
![image](https://github.com/bitburner-official/bitburner-src/assets/26317456/31553ef1-f855-44b6-b066-fa05e91c60ab)

Tested locally:
![image](https://github.com/bitburner-official/bitburner-src/assets/26317456/03d7921f-409a-4aef-b803-b96d926344c6)

Missing space between colon and number in SellDivisionModal:
![image](https://github.com/bitburner-official/bitburner-src/assets/26317456/955e0efe-442f-4490-ab8b-5bd6151dfd96)

Tested locally:
![image](https://github.com/bitburner-official/bitburner-src/assets/26317456/55c6de68-6d93-4817-a6f8-b6b618aeb209)

Missing space between "for" and number:
![image](https://github.com/bitburner-official/bitburner-src/assets/26317456/b6e47235-8641-4041-ab38-cfbfc3fa2210)

Tested locally:
![image](https://github.com/bitburner-official/bitburner-src/assets/26317456/e64d2763-010e-4d32-a70d-48b7f7427438)

cigarmemr committed:
MISC: Several typo fix in corporation modals

cigarmemr commented:
Seems that Prettier prevents change in L37. Why?

jjclark1982 commented on src/Corporation/ui/modals/SellDivisionModal.tsx:
```diff
@@ -59,8 +58,8 @@ export function SellDivisionModal(props: IProps): React.ReactElement {
-        <Typography>Cities:{getRecordKeys(divisionToSell.offices).length}</Typography>
-        <Typography>Warehouses:{getRecordKeys(divisionToSell.warehouses).length}</Typography>
+        <Typography>Cities: {getRecordKeys(divisionToSell.offices).length}</Typography>
+        <Typography>Warehouses: {getRecordKeys(divisionToSell.warehouses).length}</Typography>
```
This section might look better as a `<StatsTable>`, which automatically aligns the right column.

jjclark1982 commented:
> Seems that Prettier prevents change in L37. Why?

It enforces a maximum line length, but it usually inserts a space character `{" "}` when it does so. You can insert that yourself to get the spacing right.

cigarmemr committed:
Update SellDivisionModal.tsx

Fix Prettier failure.
Use StatsTable to align columns.
Fix misused ?? operator to show "Products" when the division is capable to produce them.

cigarmemr commented:
Thanks a lot. It seems to be better now.
For divisions with products:
![image](https://github.com/bitburner-official/bitburner-src/assets/26317456/db523cdc-bbee-4006-96ce-d9f909798798)

Without products:
![image](https://github.com/bitburner-official/bitburner-src/assets/26317456/2296a332-7cdc-44d6-85b0-52a6491d2f27)

cigarmemr committed:
Update SellDivisionModal.tsx

Add `key` to element in array.

cigarmemr committed:
Update SellDivisionModal.tsx

Finally, used Prettier and checked with format:report
